.github/workflows/integration.yaml

```diff
@@ -32,6 +32,20 @@ jobs:
           ~/go/pkg/mod
           ~/.cache/go-build
         key: ${{ runner.os }}-go-integration-${{ hashFiles('**/go.sum') }}
+    - name: Cache go-atc-router library
+      id: cache-atc
+      uses: actions/cache@v3
+      with:
+        path: |
+          /tmp/libatc_router*
+        key: ${{ runner.os }}-go-atc-router-${{ hashFiles('**/go.sum') }}
+    - name: Install cached library
+      run: |
+        if [ -e /tmp/libatc_router* ]; then
+          sudo install /tmp/libatc_router* /usr/local/lib;
+        else
+          make install-deps
+        fi
```

Comment:
This is smelly.
The CI environment should be responsible for restoring the cache if there exists one.
The script (make target) should be responsible for checking if the compiled library exists, if it is up to date, if not do its thing. The workflow on a developer's workstation and in CI should be identical.

Reply:
it was the simplest solution for dynamic linking, since the library must be "installed" in a well known place, but the standard cache action refuses to touch those directories.
in fact this is one of the big disadvantages of dynamic linking.  last time, there was some issues for fully static, but i think they can be solved a bit cleaner that this.